app/controllers/application_controller.rb

```diff
@@ -1,10 +1,16 @@
 class ApplicationController < ActionController::Base
+
   protect_from_forgery with: :exception
 
-  helper_method :cart
+  helper_method :cart, :user
 
   def cart
-    @cart ||= Cart.new(session[:cart] ||= Hash.new(0))
+    cart ||= Cart.new(session[:cart] ||= Hash.new(0))
   end
 
+  def user
```

Comment:
Is there a better way to write this so that it doesn't it the db everytime we change pages?